src/objects.c

```diff
@@ -829,7 +839,7 @@ static inline UInt IS_ON_PRINT_STACK(const ObjectsModuleState * os, Obj obj)
   if (!(FIRST_RECORD_TNUM <= TNUM_OBJ(obj)
```

Comment:
While here, maybe make the `UInt i` on line 828 be `Int i`, so we correctly handle `os->PrintObjDepth-1` being 0. This function shouldn't get called when the stack is empty, but that is one of the things that this bug caused.

Reply:
I addressed this now, albeit in a slightly different way.